Apps/UnitTests/Scripts/tests.js

```diff
@@ -101,6 +101,144 @@ describe("RequestFile", function () {
     })
 })
 
+function createSceneAndWait(callback, done) {
+    const engine = new BABYLON.NativeEngine();
+    const scene = new BABYLON.Scene(engine);
+    scene.createDefaultCameraOrLight();
+    callback(engine, scene);
+    scene.executeWhenReady(() => {
+        done();
+    })
+}
+
+describe("Materials", function() {
+    this.timeout(0);
+    it("Empty ShaderMaterial should compile", function(done) {
+        function createEmptyShaderMat() {
+            createSceneAndWait((engine, scene) => {
+                const sphere = BABYLON.MeshBuilder.CreateSphere("sphere", {diameter: 2, segments: 32}, scene);
+                const shaders = {
+                    vertexSource: `void main() {}`,
+                    fragmentSource: `void main() {}`
+                }
+                const mat = new BABYLON.ShaderMaterial("shader", scene, shaders, {});
+                sphere.material = mat;
+            }, done);
+        }
+        createEmptyShaderMat()
+    })
+    // it("GradientMaterial should compile", function(done) {
```

Comment:
might be good to group the ones that are commented out and add a comment explaining why they are commented out

Reply:
I would recommend one comment per block. This block still doesn't have a comment. It's hard to tell if it's a mistake or it's intentional.